url/src/lib.rs

```diff
@@ -219,11 +219,14 @@ mod origin;
 mod parser;
 mod path_segments;
 mod slicing;
+#[cfg(feature = "bincode")]
+use bincode::{Encode,Decode};
 
 #[doc(hidden)]
 pub mod quirks;
 
 /// A parsed URL record.
+#[cfg_attr(feature = "bincode", derive(Encode, Decode))]
```

Comment:
This custom derive will likely use the internal representation of `Url`. I do not think we wish to stabilize it as such.


Would it be possible to instead do something with the `serde` impl? Bincode supports serde.